index.js

```diff
@@ -52,6 +52,50 @@ function init() {
             process.exit(1)
         }
 
+        try {
+            // promisify APIs
+            var interfaces = {
+                  Nvim: nvim.constructor,
+                  Buffer: nvim.Buffer,
+                  Window: nvim.Window,
+                  Tabpage: nvim.Tabpage,
+                }
+
+            Object.keys( interfaces ).forEach( function( key ) {
```

Comment:
It might be more future-proof if you iterate over the keys of `nvim` directly (adding the `nvim.constructor` as necessary).